src/node_url.cc

```diff
@@ -407,29 +407,64 @@ int64_t ParseIPv4Number(const char* start, const char* end) {
   return strtoll(start, nullptr, R);
 }
 
+// https://url.spec.whatwg.org/#ipv4-number-parser
+bool IsIPv4NumberValid(const std::string_view input) {
+  if (input.empty()) {
+    return false;
+  }
+
+  // If a number starts with '0' it might be a number with base 8 or base
+  // 16. If not, checking if all characters are digits proves that it is a
+  // base 10 number.
+  if (input.size() >= 2 && input[0] == '0') {
+    if (input[1] == 'X' || input[1] == 'x') {
+      if (input.size() == 2) {
+        return true;
+      }
+
+      return input.find_first_not_of("0123456789abcdefABCDEF", 2) ==
```

Comment:
find_first_not_of() is O(n*m) - I mean, the spec doesn't say it but in practice it is.

It'd be better to use none_of() with a custom predicate (`c >= '0' && c <= '9' || c >= 'A' && c <= 'F' || c >= 'a' && c <= 'f'`) although that's a C++20 thing and I don't know if we can use that yet.

Reply:
We can use this:

```
std::all_of(input.begin(), input.end(), [](char c) { /* what bnoordhuis said*/ })
```